docs/developer/architecture/configurations.rst

```diff
@@ -73,3 +73,205 @@ As a MetalK8s expert, I can use ``kubectl`` command(s) in order to edit all
 settings that are exposed. The intent is to have a method / API that an expert
 could use, if the right CLI tool or GUI is not available or not functioning as
 expected.
+
+Design Choices
+--------------
+
+:term:`ConfigMap` is chosen as a unified data access and storage media for
+cluster and service configurations in a MetalK8s cluster based on the above
+requirements for the following reasons:
+
+* Ability to support Update operations on ConfigMap's with CLI and UI easily
+  using our already existing python kubernetes module.
+* Guarantee of adaptability and ease of changing the design and implementation
+  in cases were customer needs evolve rapidly.
+* ConfigMaps are stored in the :term:`etcd` database which is generally being
+  backed up. This ensures that user settings cannot be lost easily.
+
+How it works
+^^^^^^^^^^^^
+
+During the Bootstrap stages, when we are assertive that the K8s cluster is
+fully ready and available we could perform the following actions:
+
+  - Create and deploy ConfigMaps that hold cluster and service configurations
+    and pre-fill them with default values.
+  - Template service pods and deployments to consume configuration data
+    directly from the above deployed ConfigMaps
+
+This approach works because in a MetalK8s cluster, cluster and service
+configurations do not influence how the cluster behaves but rather the
+BootstrapConfiguration dictates the cluster behaviour. Also the creation of
+cluster and service configurations takes place before the respective consumers
+(MetalK8s addons) are deployed hence configuration data is always ready for
+consumers and not the other way round.
```

Comment:
I think you're trying to say that service configurations (and maybe others that affect the cluster **after** initial deployment) are a good fit for ConfigMaps, whereas BootstrapConfiguration needs to stay as a file, since we need it before initial deployment. It also evokes the need for us to create said CMs before deploying the configured services.

Rewording a bit, we could simply state that:
- we need special casing for some configs (what goes in BootstrapConfig)
- we can store all the rest in the API, as long as we make sure these CMs are available **before** we deploy the configured services

Also:
> cluster [...] configurations do not influence how the cluster behaves

This looks incorrect, or if it's not, then I guess those configurations are useless? :thinking: